engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/source/SqlSourceFactory.scala

```diff
@@ -0,0 +1,73 @@
+package pl.touk.nussknacker.engine.flink.table.source
+
+import pl.touk.nussknacker.engine.api.component.ProcessingMode
+import pl.touk.nussknacker.engine.api.context.ValidationContext
+import pl.touk.nussknacker.engine.api.context.transformation.{
+  DefinedEagerParameter,
+  NodeDependencyValue,
+  SingleInputDynamicComponent
+}
+import pl.touk.nussknacker.engine.api.definition._
+import pl.touk.nussknacker.engine.api.process.{BasicContextInitializer, Source, SourceFactory}
+import pl.touk.nussknacker.engine.api.{NodeId, Params}
+import pl.touk.nussknacker.engine.flink.table.extractor.SqlDataSourceConfig
+
+class SqlSourceFactory(configs: List[SqlDataSourceConfig])
+    extends SingleInputDynamicComponent[Source]
+    with SourceFactory {
+
+  override type State = Nothing
+
+  private val tableNamParamName = "Table"
+
+  private val tableNameParam: ParameterWithExtractor[String] = {
+    val param                    = ParameterWithExtractor.mandatory[String](tableNamParamName)
+    val possibleTableParamValues = configs.map(c => FixedExpressionValue(s"'${c.tableName}'", c.tableName))
+    param.copy(parameter =
```

Comment:
Ah this api is ugly :( But it is not a part of this change

Reply:
Changed it a bit to first construct a parameter and then wrap it into the `ParameterWithExtractor`